consensus/istanbul/core/request.go

```diff
@@ -33,8 +41,9 @@ func (c *core) handleRequest(request *istanbul.Request) error {
 	logger.Trace("handleRequest", "number", request.Proposal.Number(), "hash", request.Proposal.Hash())
 
 	c.current.pendingRequest = request
-	if c.state == StateAcceptRequest {
-		c.sendPreprepare(request)
+	// Must go through startNewRound to send a proposal that matches the round change certificate if not on the first round.
```

Comment:
The round change certificate does not necessarily specify a proposal, right?

Reply:
No it doesn't, but we check every message in it to see if there is a proposal we must use, and then if there isn't we can use the pendingRequest.

It's less start new round and more the logic that selects what the next pre-prepare should be (which is in start new round, but could be refactored out).